### ISSUE TITLE
Feature "Sync-File-Names"

### ISSUE THREAD

**Issue by bernd5** [dotnet/roslyn#61973]
## Summary

It would be great to have a feature which syncs type-names and file-names for single-type-files.
(It should work similar to "Sync Namespaces")

## Background and Motivation

When users copy files the name is changed to "OldFile - Copy.cs" which is fine (it is not know which name it should have).

To have synced files users have to change the name of the file and the name of the single type - which is annoying.

## Proposed Feature

The feature should be similar to "Sync Namespaces". It could may ask if the user prefers the filename or the typename.

## Alternative Designs

In a copy process of a file - the filename could be applied automatically for non-partial single type files.

**Comment by ryzngard** [comment]
I don't think this will work for just copy/paste of a file. The contents would also be the same. Instead, I think this should be counted as covered by "Rename file to <class name>": 

![image](https://user-images.githubusercontent.com/475144/178345171-14628c4e-5dde-46a5-9316-a12ba07571f4.png)

**Comment by akhera99** [comment]
Conclusion from design meeting:

We would want a "fix-all" for this scenario. A user would do their copy/paste scenarios and a fix-all would be offered to sync file names to type names granted there would be a singular type in the file.
